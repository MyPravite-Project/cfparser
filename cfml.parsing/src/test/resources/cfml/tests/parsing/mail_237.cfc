cfmail(
    to=toAddress,
    from=adminEmail,
    replyto=adminEmail,
    failto=adminEmail,
    subject="Test email",
    type="html"
) {
     writeOutput(mailBody);
};