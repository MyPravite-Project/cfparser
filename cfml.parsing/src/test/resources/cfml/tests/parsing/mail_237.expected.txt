/*===TOKENS===*/
IDENTIFIER               <cfmail>
'('                      <(>
Hidden:NEWLINE                  <>
TO                       <to>
'='                      <=>
IDENTIFIER               <toAddress>
','                      <,>
Hidden:NEWLINE                  <>
IDENTIFIER               <from>
'='                      <=>
IDENTIFIER               <adminEmail>
','                      <,>
Hidden:NEWLINE                  <>
IDENTIFIER               <replyto>
'='                      <=>
IDENTIFIER               <adminEmail>
','                      <,>
Hidden:NEWLINE                  <>
IDENTIFIER               <failto>
'='                      <=>
IDENTIFIER               <adminEmail>
','                      <,>
Hidden:NEWLINE                  <>
IDENTIFIER               <subject>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <Test email>
CLOSE_STRING             <">
','                      <,>
Hidden:NEWLINE                  <>
IDENTIFIER               <type>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <html>
CLOSE_STRING             <">
Hidden:NEWLINE                  <>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
IDENTIFIER               <writeOutput>
'('                      <(>
IDENTIFIER               <mailBody>
')'                      <)>
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (memberExpression
                (functionCall
                  (identifier cfmail)
                  (
                  (argumentList
                    (argument
                      (argumentName (identifier to))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression (unaryExpression (memberExpression (identifier toAddress))))
                        )
                      )
                    )
                    ,
                    (argument
                      (argumentName (identifier from))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression (unaryExpression (memberExpression (identifier adminEmail))))
                        )
                      )
                    )
                    ,
                    (argument
                      (argumentName (identifier replyto))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression (unaryExpression (memberExpression (identifier adminEmail))))
                        )
                      )
                    )
                    ,
                    (argument
                      (argumentName (identifier failto))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression (unaryExpression (memberExpression (identifier adminEmail))))
                        )
                      )
                    )
                    ,
                    (argument
                      (argumentName (identifier subject))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart Test email) "))
                              )
                            )
                          )
                        )
                      )
                    )
                    ,
                    (argument
                      (argumentName (identifier type))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart html) "))
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                  )
                  (compoundStatement
                    {
                    (statement
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (memberExpression
                                (functionCall
                                  (identifier writeOutput)
                                  (
                                  (argumentList
                                    (argument
                                      (startExpression
                                        (compareExpression
                                          (baseExpression (unaryExpression (memberExpression (identifier mailBody))))
                                        )
                                      )
                                    )
                                  )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                      ;
                    )
                    }
                  )
                )
              )
            )
          )
        )
      )
      ;
    )
  )
)
/*======*/
/*===DECOMPILE===*/
cfmail(to = toAddress, from = adminEmail, replyto = adminEmail, failto = adminEmail, subject = 'Test email', type = 'html')  {
writeOutput(mailBody);

  }
/*======*/